### ISSUE TITLE
Latency of AlphaTab

### ISSUE THREAD

**Issue by PedroEmanuelCosta2** [CoderLine/alphaTab#169]
# Your environment

* Version used: master version + @j4kim modification
* Platform used: JavaScript
* Rendering engine used: SVG
* Browser Name and Version: Safari Version 11.0.2 (13604.4.7.1.6) ; Chrome Version 63.0.3239.132 (Build officiel) (64 bits)
* Operating System and version (desktop or mobile): macOS High Sierra 10.13.2 (17C205)
* Link to your project: Can't share it in public because it's a private Project with a real client. (Can share a docker image with the project in it - in a private chat) 

# Expected Results

A synchronisation between a self-build metronome (with the Web Audio API) and the audio of AlphaTab. My metronome is is as accurate as a real metronome - I've made a test between a .wav file of a 60 bpm metronome and my metronome and they are synchronized. And we have detect (me and another colleague that have made some test too with a generate wav file of a metronome) a little shift in the audio of AlphaTab. I've tried to correct my metronome several time because I thought it was my fault at first. But the comparaison of my metronome and the file proved that my metronome is ok. So I think - but I can be wrong - that the shift come from the sound of AlphaTab. This shift begin directly at the start of a partition, the audio doesn't sound exactly when the cursor is on the note. I've seen that this shift is different between Safari and Chrome (Safari has a bigger shift than Chrome).

I've seen that it's possible to use your metronome but the fact is that my metronome must work with some pauses in parallel of a video (it's the continuation of @j4kim 's project). 

- Sorry for my bad english -

# Observed Results

The audio's shift of AlphaTab

# Steps to Reproduce (for bugs)

Explain here the exact steps how to reproduce the problem. Try to provide a
running sample showing the issue. If you have problem with a dedicated input file, 
please provide a download link or attach it to the issue. 
 
I can explain it to you if I give you the link of my project in private. But overall, every time I play a partition,  my metronome tone correctly every beginning of a stave. But the audio isn't synchronized with it. It's hard to explain my problem without the fact that you have my project. It would be better if I could give it to you. :)

# Possible Solution 
If you already know a solution to the problem please provide it here.

**Comment by Danielku15** [comment]
Yes, the time reported via events and the actual playback has always a small shift. The problem was reported and the reason explained here: https://github.com/CoderLine/alphaTab/issues/146

It is simply due to the fact of buffering on the audio output and hardly possible to get rid of. You find the corresponding code here:

https://github.com/CoderLine/alphaSynth/blob/6b96f3baa45620e3d066de50696e4cc0fa3bd9b0/Source/AlphaSynth.JavaScript/Main/AlphaSynthWebAudioOutput.cs#L150

You could try hooking up to the raw audio context to get the raw web audio API timings but that would mean that you will have to do the time handling on your own including the buffering, seeking, play/pause. If you find an easy solution for the problem I'd be happy to add it to alphaSynth.

**Comment by PedroEmanuelCosta2** [comment]
Okay sorry for the new issue, my bad ! I haven't found this closed issue that explain my problem. 
I've a really better vision of why there is this time report. I went crazy because I thought it was my metronome's fault ahah ! 

Thank's for all your work by the way.

Sadly it's the end of my project and I will not be working on it anymore. But thanks to you I could explain this latency. But still, I'll try in my spare time to see if I found something that could solve it yes.

Thanks again and have a good day.
